presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergCommonModule.java

```diff
@@ -168,6 +172,22 @@ public void setup(Binder binder)
         binder.bind(ConnectorPlanOptimizerProvider.class).to(IcebergPlanOptimizerProvider.class).in(Scopes.SINGLETON);
     }
 
+    @Singleton
+    @Provides
+    public StatisticsFileCache createStatisticsFileCache(IcebergConfig config, MBeanExporter exporter)
+    {
+        Cache<StatisticsFile, Map<Integer, ColumnStatistics>> delegate = CacheBuilder.newBuilder()
+                .maximumWeight(config.getMaxStatisticsFileCacheSize().toBytes())
+                .<StatisticsFile, Map<Integer, ColumnStatistics>>weigher((key, entry) ->
+                        (int) entry.values().stream().mapToLong(StatisticsUtil::getColumnStatisticsMemoryUtilization)
+                                .sum())
+                .recordStats()
+                .build();
```

Comment:
Unfortunately, there isn't much more we can configure with the guava caches unless we allow time or cache entry amount-based evictions. Time-based evictions don't make sense because the statistics files are immutable so the entries will never be stale. If old entries they fill up the cache, then you can just wait until they get evicted by newer ones. I don't see a reason to try to expire early.  Otherwise, having to modify cache size via number of entries makes it much more difficult to properly size the coordinator heap. So, I think this is best approach of the available eviction options.

I am open to ideas which gives users more flexibility. However, I think if we move towards a cache key implementation which is more granular then there I don't see much more value in configuring other parts of the cache.

Reply:
I don't think there's strong precedent within the codebase generally to support user-provided cache implementations. In general Presto makes heavy use of guava caches, but little of the exposed to the user except for some of the removal properties (size/number of entries/weight). There are at least 82 references to guava's `Cache.newBuilder` in the codebase from a quick glance.

If we had a better framework for providing a custom cache interface already, I would not be as opposed. However, It is probably a big chunk of work just support it within Iceberg. I would happily support a user-provided cache implementation in a later PR though. Let's create an issue to track it? It could be good issue to pick up for a future contributor.

Reply:
Also re-disabling cache: I would just set the new cache size configuration to 0B and it will effectively disable caching. One entry might get stored at a time, but for all intents and purposes the cache will almost always return null, causing the cache-miss logic to be executed.

Reply:
> I would happily support a user-provided cache implementation in a later PR though. Let's create an issue to track it? It could be good issue to pick up for a future contributor.

Sure, we can go back here and think about this issue if we encounter problems in production environment.

> I would just set the new cache size configuration to 0B and it will effectively disable caching.

OK, that would be fine.